builtin/logical/mysql/path_config_connection.go

```diff
@@ -40,30 +45,40 @@ func pathConfigConnection(b *backend) *framework.Path {
 
 func (b *backend) pathConnectionWrite(
 	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-	connString := data.Get("value").(string)
+	connValue := data.Get("value").(string)
 	connURL := data.Get("connection_url").(string)
+	if connURL == "" {
+		if connValue == "" {
+			return logical.ErrorResponse("provide the connection_url"), nil
```

Comment:
Pedantic, but being very direct like this can seem aggressive -- even in errors. Something like "the connection_url parameters must be supplied" is much nicer.